shared/util_test.go

```diff
@@ -40,6 +40,87 @@ func TestUrlsJoin(t *testing.T) {
 	}
 }
 
+func TestParseLXDFileHeader(t *testing.T) {
+	header := map[string][]string{
+		"X-Lxd-Uid":  {"1000"},
+		"X-Lxd-Gid":  {"1001"},
+		"X-Lxd-Mode": {"0700"},
+	}
+
+	headers, err := ParseLXDFileHeaders(header)
+	if err != nil {
+		t.Fatalf("Failed with no modify-perm: %s", err)
+		return
+	}
+
+	if headers.UID != 1000 || headers.GID != 1001 || headers.Mode != 0o700 {
+		t.Fatalf("Bad Uid (%d), Gid (%d), or Mode (%d)", headers.UID, headers.GID, headers.Mode)
+	}
+
+	if headers.ModifyExistingUID || headers.ModifyExistingGID || headers.ModifyExistingMode {
+		t.Fatalf("Incorrect `modify-perm` header")
+	}
+
+	header = map[string][]string{
+		"X-Lxd-Uid":         {"0"},
+		"X-Lxd-Gid":         {"99"},
+		"X-Lxd-Mode":        {"420"},
+		"X-Lxd-Modify-Perm": {"uid,gid,mode"},
+	}
+
+	headers, err = ParseLXDFileHeaders(header)
+	if err != nil {
+		t.Fatalf("Failed with modify-perm: %s", err)
+		return
+	}
+
+	if headers.UID != 0 || headers.GID != 99 || headers.Mode != 0o644 {
+		t.Fatalf("Bad Uid (%d), Gid (%d), or Mode (%d)", headers.UID, headers.GID, headers.Mode)
```

Comment:
In Go its generally expected and linted for that acronyms should be all upper case. So "UID" and "GID" here rather than "Uid" and "Gid".  Realise this is a just a test, but for future reference in other scenarios.